lib/controller.dart

```diff
@@ -228,6 +231,71 @@ void _computeChartBuckets(List<model.Chart> charts) {
   logger.debug('Computed chart buckets ${charts}');
 }
 
+String _generateLegendLabelFromFilters(Map<String, String> filters) {
```

Comment:
consider moving the generate chart config functions to a `chart_helpers.dart` file - they're pretty self-contained and not very related to the core logic of the tool, as well as probably being somewhat closer to the view than the controller.

Reply:
That makes sense, all the chart functions are interlinked and moved to chart_helpers.dart